packages/graph/hash_graph/lib/graph/src/api/rest/link_type.rs

```diff
@@ -0,0 +1,167 @@
+        create_link_type,
+        get_link_type,
+        // get_link_type_many,
+        update_link_type
+    ),
+    components(CreateLinkTypeRequest, UpdateLinkTypeRequest, AccountId, QualifiedLinkType),
+    tags(
+        (name = "LinkType", description = "Link type management API")
+    )
+)]
+pub struct LinkTypeResource;
+
+impl RoutedResource for LinkTypeResource {
+    /// Create routes for interacting with link types.
+    fn routes<D: Datastore>() -> Router {
+        // TODO: The URL format here is preliminary and will have to change.
+        Router::new().nest(
+            "/link-type",
+            Router::new()
+                .route("/", post(create_link_type::<D>).put(update_link_type::<D>))
+                .route("/:version_id", get(get_link_type::<D>)),
+        )
+    }
+}
+
+#[derive(Serialize, Deserialize, Component)]
+struct CreateLinkTypeRequest {
+    #[component(value_type = Any)]
+    schema: LinkType,
+    account_id: AccountId,
+}
+
+#[utoipa::path(
+    post,
+    path = "/link-type",
+    request_body = CreateLinkTypeRequest,
+    tag = "LinkType",
+    responses(
+      (status = 201, content_type = "application/json", description = "Link type created successfully", body = QualifiedLinkType),
+      (status = 422, content_type = "text/plain", description = "Provided request body is invalid"),
+
+      (status = 409, description = "Unable to create link type in the datastore as the base link type ID already exists"),
+      (status = 500, description = "Datastore error occurred"),
+    ),
+    request_body = CreateLinkTypeRequest,
+)]
+async fn create_link_type<D: Datastore>(
+    body: Json<CreateLinkTypeRequest>,
+    datastore: Extension<D>,
+) -> Result<Json<Qualified<LinkType>>, StatusCode> {
+    let Json(body) = body;
+    let Extension(datastore) = datastore;
+
+    datastore
+        .clone()
+        .create_link_type(body.schema, body.account_id)
+        .await
+        .map_err(|report| {
+            if report.contains::<BaseUriAlreadyExists>() {
+                return StatusCode::CONFLICT;
+            }
+
+            // Insertion/update errors are considered internal server errors.
+            StatusCode::INTERNAL_SERVER_ERROR
+        })
+        .map(Json)
+}
+
+#[utoipa::path(
+    get,
+    path = "/link-type/{versionId}",
+    tag = "LinkType",
+    responses(
+        (status = 200, content_type = "application/json", description = "Link type found", body = QualifiedLinkType),
+        (status = 422, content_type = "text/plain", description = "Provided version_id is invalid"),
+
+        (status = 404, description = "Link type was not found"),
+        (status = 500, description = "Datastore error occurred"),
+    ),
+    params(
+        ("versionId" = Uuid, Path, description = "The version ID of link type"),
+    )
+)]
+async fn get_link_type<D: Datastore>(
+    version_id: Path<Uuid>,
+    datastore: Extension<D>,
+) -> Result<Json<Qualified<LinkType>>, impl IntoResponse> {
```

Comment:
Yet another oversight from the original PR introducing these endpoints, but `impl IntoResponse` isn't necessary in any of the endpoints right now as `StatusCode` would suffice. But `impl IntoResponse` is probably what we want everywhere, though. Just flagging